### ISSUE TITLE
Different behavior between [x86_64-darwin19] and [x86_64-linux] on hashes with nil values

### ISSUE THREAD

**Issue by spyrbri** [ruby/psych#467]
Ruby:
```
# ruby -v
ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x86_64-linux]
```

Yaml:
```
# gem info yaml

*** LOCAL GEMS ***

yaml (0.1.0)
    Authors: Aaron Patterson, SHIBATA Hiroshi
    Homepage: https://github.com/ruby/yaml
    License: BSD-2-Clause
    Installed at (default): /usr/local/lib/ruby/gems/2.7.0

    YAML Ain't Markup Language
```

Code:
```ruby
# irb
irb(main):001:0> require "yaml"
=> true
irb(main):002:0> tree = {"en"=>{"empty"=>nil}}
=> {"en"=>{"empty"=>nil}}
irb(main):003:0>  Psych.dump(tree)
=> "---\nen:\n  empty: \n"
```

-----------------

Ruby:
```
# ruby -v
ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x86_64-darwin19]
```

Yaml:
```
gem info yaml

*** LOCAL GEMS ***

yaml (0.1.0)
    Authors: Aaron Patterson, SHIBATA Hiroshi
    Homepage: https://github.com/ruby/yaml
    License: BSD-2-Clause
    Installed at (default): /.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0

    YAML Ain't Markup Language
```

Code:
```ruby
# irb
irb(main):001:0> require "yaml"
=> true
irb(main):002:0> tree = {"en"=>{"empty"=>nil}}
=> {"en"=>{"empty"=>nil}}
irb(main):003:0> Psych.dump(tree)
=> "---\nen:\n  empty:\n"
```

Notice the space after `empty:` -> `"---\nen:\n  empty: \n"` vs `"---\nen:\n  empty:\n"`

**Comment by perlpunk** [comment]
That is probably because of different libyaml versions.
The trailing space was fixed in libyaml 0.2.5: https://github.com/yaml/libyaml/pull/186

**Comment by spyrbri** [comment]
I am closing this since the gem is ok.

The prolbem is that we have to have the same version of `libyaml` on our dev envs and CI.

Thank you @perlpunk
